module/zfs/dsl_crypt.c

```diff
@@ -2254,6 +2291,14 @@ dsl_crypto_populate_key_nvlist(dsl_dataset_t *ds, nvlist_t **nvl_out)
 	if (ret != 0)
 		goto error;
 
+	ret = zap_lookup(mos, dckobj, DSL_CRYPTO_KEY_VERSION, 8, 1, &version);
+	if (ret != 0 || version != ZIO_CRYPT_KEY_CURRENT_VERSION) {
+		/* We don't support raw sends of legacy on-disk formats */
```

Comment:
Can you add expand on why this isn't supported.  I'd also pull the block comment up above the `zap_lookup()` call.